Feature/create data base

romanchukegor opened the pull request.

Commit:
feat: create data base

created data base and 3 schemes for storing users, appointments and tokens

Commit:
feat: created data base and schemes

created data base, created 3 schemes for storing users, tokens and appointments

Commit:
feat: created data base and schemes

created data base, created 3 schemes for storing users, appointments and tokens

baklanovairina commented on .env:
```diff
@@ -1,2 +1,2 @@
-URL="mongodb+srv://egor:D29WrwXmCYyzUTAW@cluster0.dtmjl.mongodb.net/appointment
-
+URL="mongodb+srv://egor:gfbV37EHW7dYtqRd@cluster0.jpkjqek.mongodb.net/hospital"
```
why is env here?

baklanovairina replied:
why is env here?

baklanovairina commented on app.js:
```diff
@@ -1,11 +1,26 @@
-const express = require('express')
-const app = express()
-const port = 8000
+require("dotenv").config();
```
where is the config?

baklanovairina commented on app.js:
```diff
@@ -1,11 +1,26 @@
-  res.send('Hello World!')
-})
+const app = express();
+const PORT = process.env.PORT || 8000;
```
it should be in config
and why is the alternative port the same as the main one?

baklanovairina commented on app.js:
```diff
@@ -1,11 +1,26 @@
+
+const start = async () => {
+  try {
+    await mongoose.connect(process.env.URL, {
```
process.env.URL  in config

baklanovairina commented on app.js:
```diff
@@ -1,11 +1,26 @@
+    });
+    app.listen(PORT, () => console.log(`server started on port ${PORT}`));
+  } catch (error) {
+    console.log(error)
```
where is process.exit?

baklanovairina commented on src/models/appointment-model.js:
```diff
@@ -0,0 +1,23 @@
+  },
+  date: {
+    type: Date,
+    default: new Date().toLocaleDateString("ru"),
```
Why is the default date?

baklanovairina commented on src/models/appointment-model.js:
```diff
@@ -0,0 +1,23 @@
+    default: new Date().toLocaleDateString("ru"),
+    required: true,
+  },
+  сomplaints: {
```
сomplaint

baklanovairina commented on src/models/token-model.js:
```diff
@@ -0,0 +1,8 @@
+
+const TokenSchema = new Schema({
+  user: { type: Schema.Types.ObjectId, ref: "User" },
+  refreshToken: { type: String, required: false },
```
cod style
and why false?

baklanovairina commented on src/models/token-model.js:
```diff
@@ -0,0 +1,8 @@
+const { Schema, model } = require("mongoose");
+
+const TokenSchema = new Schema({
+  user: { type: Schema.Types.ObjectId, ref: "User" },
```
cod style

baklanovairina commented on src/models/appointment-model.js:
```diff
@@ -0,0 +1,23 @@
+  },
+});
+
+module.exports = model("Appointment", AppointmentSchema);
```
"AppointmentS"

baklanovairina commented on src/models/token-model.js:
```diff
@@ -0,0 +1,8 @@
+  refreshToken: { type: String, required: false },
+});
+
+module.exports = model("Token", TokenSchema);
```
"TokenS"

baklanovairina commented on src/models/user-model.js:
```diff
@@ -0,0 +1,8 @@
+  password: { type: String, unique: true, required: true },
+});
+
+module.exports = model("User", UserSchema);
```
"UserS"

baklanovairina commented on src/models/user-model.js:
```diff
@@ -0,0 +1,8 @@
+const { Schema, model } = require("mongoose");
+
+const UserSchema = new Schema({
+  login: { type: String, unique: true, required: true },
```
cod style

baklanovairina commented on src/models/user-model.js:
```diff
@@ -0,0 +1,8 @@
+
+const UserSchema = new Schema({
+  login: { type: String, unique: true, required: true },
+  password: { type: String, unique: true, required: true },
```
cod style
are you sure the password must be unique?

baklanovairina commented on src/models/appointment-model.js:
```diff
@@ -0,0 +1,23 @@
+  сomplaints: {
+    type: String,
+    required: true,
+  },
```
you also need a userId to link the user to his appointments

Commit:
fix(common): corrected code style and structure of application

corrected indents and tabs, created config file, rename some elements, added .env in gitignore, remove service folder

Commit:
fix: deleted unique

Commit:
fix: deleted .env in repository
